test/jest-runner.js

```diff
@@ -0,0 +1,210 @@
+const path = require('path');
+const wrapper = require('@playwright/jest-wrapper');
+const DefaultTestRunner = require('../utils/testrunner');
+const {TestRunner, TestRun} = require('../utils/testrunner/TestRunner');
+const {Environment} = require('../utils/testrunner/Test');
+const utils = require('./utils');
+const fs = require('fs');
+const pirates = require('pirates');
+const babel =require('@babel/core');
+const testRunnerInfo = makeTestRunnerInfo();
+
+
+module.exports = wrapper.createJestRunner(async ({path: filePath}) => {
+  const config = require('./test.config');
+  const spec = config.specs.find(spec => {
+    return spec.files.some(f => path.join(__dirname, f) === filePath);
+  });
+  if (!spec) {
+    console.error('cannot find spec for', filePath);
+    return [];
+  }
+  const {testRunner, browserInfo} = testRunnerInfo;
+  testRunner.collector()._tests = [];
+  for (const [key, value] of Object.entries(testRunner.api()))
+    global[key] = value;
+  for (const {browserEnvironment, browserTypeEnvironment, browserName, browserType, pageEnvironment, launchOptions} of browserInfo) {
+    const suiteName = { 'chromium': 'Chromium', 'firefox': 'Firefox', 'webkit': 'WebKit' }[browserName];
+    describe(suiteName, () => {
+      // In addition to state, expose these two on global so that describes can access them.
+      global.browserType = browserType;
+      global.HEADLESS = !!launchOptions.headless;
+
+      testRunner.collector().useEnvironment(browserTypeEnvironment);
+      const skip = spec.browsers && !spec.browsers.includes(browserName);
+      (skip ? xdescribe : describe)(spec.title || '', () => {
+        for (const e of spec.environments || ['page']) {
```

Comment:
I'd suggest to only run tests from the page environment here, so that we can iterate on the user-like experience. wdyt?

Reply:
Talked about it offline: